openedx/frontend-platform
# feat: UI slot component with the basic config

**Description:**

This PR the basic `PluggableComponent` which will allow to reuse it where we need it and an util called `contextFactory` will help us to have a context provider that we will use thro a reducer to share data between components or even PluggableComponent if you rather. The main idea is to have those to be reusable in any MFE that way we don't need to setup them. 

This PR depends on this one: https://github.com/openedx/frontend-build/pull/518 

#### How to test it

1. Install dependencies
```
nvm use && npm install
```
2. Start the frontend-build
```
npm start
```
3. You should be able to see this following plugin which is only for test purpuses:
<img width="1409" alt="Screen Shot 2024-03-13 at 1 32 01 PM" src="https://github.com/openedx/frontend-platform/assets/134975835/cc8b7447-b372-4c1f-9855-d1cb41c9b399">




**Merge checklist:**

- [x] Consider running your code modifications in the included example app within `frontend-platform`. This can be done by running `npm start` and opening http://localhost:8080.
- [ ] Consider testing your code modifications in another local micro-frontend using local aliases configured via [the `module.config.js` file in `frontend-build`](https://github.com/openedx/frontend-build#local-module-configuration-for-webpack).
- [ ] Verify your commit title/body conforms to the conventional commits format (e.g., `fix`, `feat`) and is appropriate for your code change. Consider whether your code is a breaking change, and modify your commit accordingly.

**Post merge:**

- [ ] After the build finishes for the merged commit, verify the new release has been pushed to [NPM](https://www.npmjs.com/package/@edx/frontend-platform).
